core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -30,6 +30,7 @@ import scala.util.Properties
 import org.apache.commons.lang3.StringUtils
 import org.apache.hadoop.fs.Path
 import org.apache.hadoop.security.UserGroupInformation
+import org.apache.hadoop.yarn.conf.YarnConfiguration
```

Comment:
Yea it looks like this should require `hadoop-yarn-api`.  I'll add this dependency to `core` unless you object, in which case I suppose I could just use the raw string instead of `YarnConfiguration.RM_PRINCIPAL`, but that seems hacky.

FYI, we originally talked about placing the code below into the Mesos scheduler, but that seems to be too late.  The `SparkContext` constructor creates a copy of the configuration, so we need to set the required YARN property before `SparkContext` is created, which means before user code gets run, which probably means somewhere in SparkSubmit.